drivers/can/stm32_can.c

```diff
@@ -0,0 +1,765 @@
+/*
+ * Copyright (c) 2018 Alexander Wachter
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <clock_control/stm32_clock_control.h>
+#include <clock_control.h>
+#include <misc/util.h>
+#include <string.h>
+#include <kernel.h>
+#include <board.h>
+#include <errno.h>
+#include <can.h>
+#include "stm32_ll_can.h"
+
+#define SYS_LOG_LEVEL CONFIG_SYS_LOG_CAN_LEVEL
+#include <logging/sys_log.h>
+
+static const struct can_driver_api can_api_funcs = {
```

Comment:
We always declare the driver API at the end of the driver code, before the DEVICE_INIT() macro call.

Follow that, and then remove the function declarations in stm32_can.h